scanner/src/main/kotlin/scanners/fossid/FossIdNamingProvider.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.scanners.fossid
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+import org.ossreviewtoolkit.utils.log
+
+/**
+ * This class provides names for projects and scans when the FossID scanner creates them, following a given pattern.
+ * If one or both patterns is null, a default naming convention is used.
+ *
+ * [namingScanPattern] and [namingProjectPattern] are patterns describing the name using variables, e.g. "$var1_$var2".
+ * Variable values are given in the map [namingConventionVariables].
+ *
+ * There also are builtin variables. Builtin variables are are prefixed in the pattern with "#" e.g. "$var1_#builtin".
+ * Available builtin variables:
+ * * **projectBaseCode**: The name of the project (i.e. the part of the URL before .git).
```

Comment:
Would it make sense to have the current timestamp as a built-in variable?

Reply:
YAGNI but ok :)

Reply:
I think it is already used for the default pattern; so if you just slightly wanted to adapt the default pattern, you would need such a variable.